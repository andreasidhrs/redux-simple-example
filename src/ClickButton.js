import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ClickButton extends Component {
  static propTypes = {
    startClicks: PropTypes.number
  };

  static defaultProps = {
    startClicks: 0
  }

  state = {
    totalClicks: this.props.startClicks
  }

  clickHandler = () => {
    const totalClicks = (this.state.totalClicks + 1);
    this.setState({ totalClicks });
  }

  render() {
    const { totalClicks } = this.state;
    return (<button onClick={this.clickHandler}>count {totalClicks}</button>);
  }
}
