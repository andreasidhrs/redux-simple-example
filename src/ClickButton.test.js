import React from 'react';
import ClickButton from './ClickButton';
import renderer from 'react-test-renderer';

describe('ClickButton', () => {

  beforeEach(() => console.log('before'));
  afterEach(() => console.log('after'));

  it('should render correctly', () => {
    const tree = renderer
      .create(<clickbutton/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
