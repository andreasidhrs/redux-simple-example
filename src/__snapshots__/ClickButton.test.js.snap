// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickButton should render correctly 1`] = `<clickbutton />`;
